asllib/doc/AbstractSyntax.tex

```diff
@@ -692,7 +692,7 @@ \subsection{Abbreviated Rule Notation for AST Builders}
 \inferrule{}{
 {
 \begin{array}{r}
-  \buildstmt(\Nstmt(\Twhile, \punnode{\Nexpr}, \Tdo, \punnode{\Nstmtlist}, \Tend))
+  \buildstmt(\Nstmt(\Twhile, \punnode{\Nexpr}, \Tdo, \punnode{\Nstmtlist}, \Tend, \Tsemicolon))
```

Comment:
I'm guessing you search for each occurrence of `\Tend` and added `, `\Tsemicolon` to make sure you haven't missed any, right?

Reply:
Something like that. Some of them didn't need a comma following the `\Tend`, they needed `\parsesep` but I did double check that every occurrence of `\Tend` is followed by a semicolon in some way. (All inside the `asllib/doc` folder except for the `\Tend` definition in `asllib/doc/ASLmacros.tex`). I also did a quick search for `"end"` in the rendered pdf and it seems to have been added everywhere.